aplib.net-demo/Assets/Scripts/Doors/DoorCollision.cs

```diff
@@ -2,26 +2,32 @@
 
 /// <summary>
 /// This class handles collisions with the player and makes sure that the parent object (the door) disappears/opens when
-/// the player is in range and the prerequisites are met (the right key, or in this case the right player ID).
-/// Set the ID in the editor for the collider
+/// the player is in range and the prerequisites are met (the right key).
 /// </summary>
 public class DoorCollision : MonoBehaviour
 {
+    /// <summary>
+    /// The number of doors that have been spawned in the level so far. This is used to give each door a unique ID.
+    /// </summary>
+    private static int s_numberOfDoors;
+
     /// <summary>The unique ID of the door, to check whether the player has the right key/ID to open the door</summary>
-    public int doorId;
-    /// <summary>the door object this script is attached to</summary>
-    [SerializeField] private GameObject parent;
+    [SerializeField] private int _doorId;
```

Comment:
I wouldn't make this private, because it needs to be accessed to check whether or not the player has the right key

Reply:
I would argue that the field shouldn't even be serialized.

Reply:
1. that's a fair point, but I think it's equally as logical to implement it somewhere else, the door, the key, and the player turning the key have equal impact when opening a door
2. The ID is a representation of the notches on the key/inside the lock so in a way it does
3. This is one solution, but again I think the door is not more responsible for deciding if it opens than the key or player
4. As long as the door has a function to be removed, I don't think it makes it more difficult at all to open it in other ways. If I were to make an item that opens the next 3 doors, I would create the logic there and just call the destroydoor function where needed

Reply:
1. They might have equal impact in the sense that they all play a role in opening a door. However, they don't have equal responsibilities.
2. Correct; "inside the lock". A real-life door uses an 'ID' in the same way a virtual door should: for the inner logic. As a frequent door opener myself, I can say I have never seen the insides of a lock. So `private` seems the desirable accessibility for the ID.
3. Of course the exact responsibility is debatable, but consider real life again. You cannot move through a door if the lock is broken. The key, or me as a door opener, cannot decide to 'open the door anyway', despite the broken lock. A broken key, however, is no problem in most cases. Just grab another key. Or we do not have another key; pity, but then we just have to accept that we cannot open the door. We have no power to open the door at that point. But the door is still working as it should, waiting for the right key.
4. It doesn't necessarily make it more difficult to open the door in other ways. In fact, I think it makes it easier to open the door in other ways if you give the door a public function to be removed. However, I'm convinced this will cause tightly coupled code, illogical code architecture, etc., giving rise to bugs, increasing the difficulty to implement new features, and overall just making the code harder to read and understand.

Reply:
I agree with @JensSteenmetz on this. I will remove the `SerializeField` attribute.